From c23e1dc22deb495561cffb877edb2746b740a1fa Mon Sep 17 00:00:00 2001
From: Jussi Kukkonen <jussi.kukkonen@intel.com>
Date: Wed, 5 Jul 2017 11:07:05 +0300
Subject: [PATCH 2/2] vulkan: Use the generated version of vkconfig.h

Build fails in ext/vulkan/xcb and ext/vulkan/wayland when:
* building from tarball
* building out-of-tree
* Only one WSI integration (xcb or wayland) is enabled by configure.ac
This is because vkconfig.h from source directory gets used instead
of the generated one.

Add the correct build directory to "-I". Use angle bracket
include in vkapi.h so that it actually looks in the include search
path instead of defaulting to the same (source tree) directory.

Signed-off-by: Jussi Kukkonen <jussi.kukkonen@intel.com>
Upstream-Status: Submitted [https://bugzilla.gnome.org/show_bug.cgi?id=784539]
---
 ext/vulkan/vkapi.h             | 2 +-
 ext/vulkan/wayland/Makefile.am | 1 +
 ext/vulkan/xcb/Makefile.am     | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/vulkan/vkapi.h b/ext/vulkan/vkapi.h
index e9c23aa92..a37c29d0f 100644
--- a/ext/vulkan/vkapi.h
+++ b/ext/vulkan/vkapi.h
@@ -23,7 +23,7 @@
 
 #define VK_PROTOTYPES
 
-#include "vkconfig.h"
+#include <vkconfig.h>
 #include "vk_fwd.h"
 #include "vkmacros.h"
 
diff --git a/ext/vulkan/wayland/Makefile.am b/ext/vulkan/wayland/Makefile.am
index f92d85e2c..10cfb70e6 100644
--- a/ext/vulkan/wayland/Makefile.am
+++ b/ext/vulkan/wayland/Makefile.am
@@ -14,6 +14,7 @@ noinst_HEADERS = \
 
 libgstvulkan_wayland_la_CFLAGS = \
 	-I$(top_srcdir)/gst-libs \
+	-I$(top_builddir)/ext/vulkan \
 	-I$(top_srcdir)/ext/vulkan \
 	-I$(top_builddir)/gst-libs \
 	$(GST_PLUGINS_BASE_CFLAGS) \
diff --git a/ext/vulkan/xcb/Makefile.am b/ext/vulkan/xcb/Makefile.am
index 7debcff9e..b5103551b 100644
--- a/ext/vulkan/xcb/Makefile.am
+++ b/ext/vulkan/xcb/Makefile.am
@@ -14,6 +14,7 @@ noinst_HEADERS = \
 
 libgstvulkan_xcb_la_CFLAGS = \
 	-I$(top_srcdir)/gst-libs \
+	-I$(top_builddir)/ext/vulkan \
 	-I$(top_srcdir)/ext/vulkan \
 	-I$(top_builddir)/gst-libs \
 	$(GST_PLUGINS_BASE_CFLAGS) \
-- 
2.13.2

